flink-yarn/src/main/java/org/apache/flink/yarn/YarnLocalResourceDescriptor.java

```diff
@@ -38,27 +39,41 @@
 class YarnLocalResourceDescriptor {
 
 	private static final String STRING_FORMAT = "YarnLocalResourceDescriptor{" +
-		"key=%s, path=%s, size=%d, modificationTime=%d, visibility=%s}";
+		"key=%s, path=%s, size=%d, modificationTime=%d, visibility=%s, type=%s}";
 	private static final Pattern LOCAL_RESOURCE_DESC_FORMAT = Pattern.compile("YarnLocalResourceDescriptor\\{" +
-		"key=(\\S+), path=(\\S+), size=([\\d]+), modificationTime=([\\d]+), visibility=(\\S+)}");
+		"key=(\\S+), path=(\\S+), size=([\\d]+), modificationTime=([\\d]+), visibility=(\\S+), type=(\\S+)}");
 
 	private final String resourceKey;
 	private final Path path;
 	private final long size;
 	private final long modificationTime;
 	private final LocalResourceVisibility visibility;
+	private final LocalResourceType resourceType;
 
 	YarnLocalResourceDescriptor(
 			String resourceKey,
 			Path path,
 			long resourceSize,
 			long modificationTime,
-			LocalResourceVisibility visibility) {
+			LocalResourceVisibility visibility,
+			LocalResourceType resourceType
+			) {
 		this.resourceKey = checkNotNull(resourceKey);
 		this.path = checkNotNull(path);
 		this.size = resourceSize;
 		this.modificationTime = modificationTime;
 		this.visibility = checkNotNull(visibility);
+		this.resourceType = resourceType;
+	}
+
+	YarnLocalResourceDescriptor(
+		String resourceKey,
```

Comment:
Add one more tab of indentation so that the separation is clear between the args and the body and the `) {` can go to the same line as the last argument.